.github/workflows/ci.yml

```diff
@@ -0,0 +1,51 @@
+name: Run Tests
+
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+
+jobs:
+  build_test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        perl:
+          - '5.38'
+          - '5.36'
+          - '5.34'
+          - '5.32'
+          - '5.30'
+          - '5.28'
+          - '5.26'
+          - '5.24'
+          - '5.22'
+    name: Perl ${{ matrix.perl }}
+
+    steps:
+      - uses: actions/checkout@v3
```

Comment:
µ-optimization: as far as you don't need history you can use shallow copy
```
with:
  fetch-depth: 1
```

Reply:
This is already the default: <https://github.com/actions/checkout/blob/v3.5.3/README.md>.

>     # Number of commits to fetch. 0 indicates all history for all branches and tags.
>     # Default: 1
>     fetch-depth: ''

Reply:
btw, I just wrote reusable workflow for PP perl modules, maybe not perfect (yet) ... but already has feature of automatically propagate newer perl version.

Take a look at https://github.com/happy-barney/github-workflows maybe it will be worth to use it here as well

Reply:
Cool! I do something similar [here with BioPerl](https://github.com/bioperl/bioperl-live/blob/febf38e528ec4db6c6fd96eb2ed6faef0445dd7f/.github/workflows/ci.yml#L54-L75) to choose a couple (min, median, max) to test against, but using the tooling that `shogo82148/actions-setup-perl@v1` provides.